optimum/neuron/modeling_seq2seq.py

```diff
@@ -0,0 +1,912 @@
+
+        save_dir = TemporaryDirectory()
+        save_dir_path = Path(save_dir.name)
+
+        main_export(
+            model_name_or_path=model_id,
+            output=save_dir_path,
+            compiler_kwargs=compiler_kwargs,
+            task=task,
+            dynamic_batch_size=dynamic_batch_size,
+            cache_dir=cache_dir,
+            trust_remote_code=trust_remote_code,
+            subfolder=subfolder,
+            revision=revision,
+            force_download=force_download,
+            local_files_only=local_files_only,
+            use_auth_token=use_auth_token,
+            do_validation=False,
+            **kwargs_shapes,
+        )
+
+        return cls._from_pretrained(
+            model_id=save_dir_path,
+            config=config,
+            model_save_dir=save_dir,
+        )
+
+    def _save_config(self, save_directory):
+        save_directory = Path(save_directory)
+        self.configs[ENCODER_NAME].save_pretrained(save_directory / ENCODER_NAME)
+        self.configs[DECODER_NAME].save_pretrained(save_directory / DECODER_NAME)
+        combined_config = self._combine_encoder_decoder_config(
+            encoder_config=self.configs[ENCODER_NAME],
+            decoder_config=self.configs[DECODER_NAME],
+        )
+        combined_config.save_pretrained(save_directory)
+
+    def _combine_encoder_decoder_config(self, encoder_config: "PretrainedConfig", decoder_config: "PretrainedConfig"):
+        encoder_neuron_config = encoder_config.neuron
+        decoder_neuron_config = decoder_config.neuron
+        combined_config = copy.deepcopy(encoder_config)
+
+        encoder_neuron_config["encoder_input_names"] = encoder_neuron_config.pop("input_names")
+        encoder_neuron_config["encoder_output_names"] = encoder_neuron_config.pop("output_names")
+        decoder_neuron_config["decoder_input_names"] = decoder_neuron_config.pop("input_names")
+        decoder_neuron_config["decoder_output_names"] = decoder_neuron_config.pop("output_names")
+
+        encoder_neuron_config.update(decoder_neuron_config)
+        encoder_neuron_config.pop("model_type")
+        combined_config.__setattr__("neuron", encoder_neuron_config)
+
+        return combined_config
+
+    def can_generate(self):
+        logger.warning(
+            "NeuronModelForConditionalGeneration is an abstract class and is not meant to be used for generation. Please use NeuronModelForSeq2SeqLM instead."
+        )
+        return False
+
+
+class NeuronModelForSeq2SeqLM(NeuronModelForConditionalGeneration, NeuronGenerationMixin):
+    auto_model_class = AutoModelForSeq2SeqLM
+    main_input_name = "input_ids"
+
+    def forward(
+        self,
+        attention_mask: Optional[torch.FloatTensor] = None,
+        decoder_input_ids: Optional[torch.LongTensor] = None,
+        decoder_attention_mask: Optional[torch.BoolTensor] = None,
+        encoder_outputs: Optional[Tuple[Tuple[torch.Tensor]]] = None,
+        beam_scores=None,
+        **kwargs,
```

Comment:
Why?
Usually I try to avoid that as much as possible.

Reply:
Yeah, indeed, it's that `prepare_inputs_for_generation` will prepare some inputs actually not traced in the model so will be left useless (all in `kwargs`), I will override the method to forward exactly inputs needed here.